### PR TITLE
switch to reqwest

### LINKED ISSUE #303: Use reqwest as client library for WASM browser support
Having a quick glance to the dependencies I see you depend on hyper directly, would it be a big effort to use reqwest instead? is one of the most used http clients this days and provides a nicer API, on top of that they already went through the effort of making it work with web assembly which seems like a major effort that this project doesn't look very motivated to take. Reqwest would act as the abstraction layer that makes the switch from a hyper backend to browser fetch at compile time.  
Nicer http api plus wasm support, sounds like a nice deal right?

### PR DESCRIPTION
**Pull request by clouds56**
fixes #303

### CODE REVIEW

File: src/transports/http.rs [hunk 1]
```diff
@@ -18,105 +22,59 @@ use std::{
...
-    url: hyper::Uri,
+    url: reqwest::Url,
     basic_auth: Option<HeaderValue>,
-    client: Client,
+    client: reqwest::Client,
 }
 
 impl Http {
     /// Create new HTTP transport connecting to given URL.
+    #[allow(unused_mut)]
```
Comment (tomusdrw) [inline src/transports/http.rs:51]: Couldn't this be moved right above of `let mut client_builder` line?

**Review by tomusdrw — COMMENTED** [review]
That's awesome, thanks a lot! Could you please bump the version as well?

File: Cargo.toml [hunk 2]
```diff
@@ -66,11 +64,13 @@ tokio-stream = { version = "0.1", features = ["net"] }
...
-http = ["_http_base", "hyper-proxy/tls"]
-http-tls = ["hyper-tls", "http"]
-http-rustls = ["hyper-rustls", "_http_base", "hyper-proxy/rustls-webpki"]
+wasm = ["js-sys", "wasm-bindgen", "wasm-bindgen-futures", "futures-timer/wasm-bindgen", "rand", "getrandom"]
+eip-1193 = ["wasm"]
+_http_base = ["reqwest", "bytes", "url", "base64", "headers"]
+http = ["_http_base"]
+http-tls = ["http", "reqwest/default-tls"]
+http-native-tls = ["http", "reqwest/native-tls"]
+http-rustls-tls = ["http", "reqwest/rustls-tls"]
```
Comment (tomusdrw) [inline Cargo.toml:73]: Would you mind adding two additional entries in `.github/workflows/rust.yml` to test these features work in isolation?

**Review by tomusdrw — COMMENTED** [review]
Thanks again, I've just realised that it would be good to test newly introduced features on CI as well.

File: Cargo.toml (hunk 2 above)
Comment (clouds56) [reply Cargo.toml:73]: I'm not familiar with Github action, tried and plz review.

**Comment by clouds56** [comment]
Shall we run ci first?

**Review by tomusdrw — APPROVED** [review]

**Pull request merged**